src/System.Configuration/src/System/Configuration/DpapiProtectedConfigurationProvider.cs

```diff
@@ -0,0 +1,137 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Specialized;
+using System.Security.Cryptography;
+using System.Text;
+using System.Xml;
+
+namespace System.Configuration
+{
+    public sealed class DpapiProtectedConfigurationProvider : ProtectedConfigurationProvider
+    {
+        private const int CRYPTPROTECT_UI_FORBIDDEN = 0x1;
+        private const int CRYPTPROTECT_LOCAL_MACHINE = 0x4;
+        private bool _useMachineProtection = true;
+        private string _keyEntropy;
+
+        public override XmlNode Decrypt(XmlNode encryptedNode)
+        {
+            if (encryptedNode.NodeType != XmlNodeType.Element ||
+                encryptedNode.Name != "EncryptedData")
+            {
+                throw new ConfigurationErrorsException(SR.DPAPI_bad_data);
+            }
+
+            XmlNode cipherNode = TraverseToChild(encryptedNode, "CipherData", false);
+            if (cipherNode == null)
+                throw new ConfigurationErrorsException(SR.DPAPI_bad_data);
+
+            XmlNode cipherValue = TraverseToChild(cipherNode, "CipherValue", true);
+            if (cipherValue == null)
+                throw new ConfigurationErrorsException(SR.DPAPI_bad_data);
+
+            string encText = cipherValue.InnerText;
+            if (encText == null)
+                throw new ConfigurationErrorsException(SR.DPAPI_bad_data);
+
+            string decText = DecryptText(encText);
+            XmlDocument xmlDocument = new XmlDocument();
+            xmlDocument.PreserveWhitespace = true;
+            xmlDocument.LoadXml(decText);
+            return xmlDocument.DocumentElement;
+        }
+
+        public override XmlNode Encrypt(XmlNode node)
+        {
+            string text = node.OuterXml;
+            string encText = EncryptText(text);
+            string pre = @"<EncryptedData><CipherData><CipherValue>";
+            string post = @"</CipherValue></CipherData></EncryptedData>";
+            string xmlText = pre + encText + post;
+
+            XmlDocument xmlDocument = new XmlDocument();
+            xmlDocument.PreserveWhitespace = true;
+            xmlDocument.LoadXml(xmlText);
+            return xmlDocument.DocumentElement;
+        }
+
+        private string EncryptText(string clearText)
+        {
+            if (clearText == null || clearText.Length < 1)
+                return clearText;
+
+            byte[] inputData = PrepareDataBlob(clearText);
+            byte[] entropyData = PrepareDataBlob(_keyEntropy);
+
+            byte[] encryptedData = ProtectedData.Protect(
```

Comment:
The only difference that I see between this and Desktop is in what exceptions could be thrown.

Desktop throws `Marshal.ThrowExceptionForHR`, which will probably turn an unloaded profile (which a code comment suggests is indicated by `ERROR_FILE_NOT_FOUND`) into a `FileNotFoundException` (https://msdn.microsoft.com/en-us/library/9ztbc5s1(v=vs.110).aspx) and a COMException for most cryptographic errors (`NTE_*`).

This will throw a CryptographicException for all things that go wrong (some with the HR, some without).

My personal opinion is that this is fine.